joomla/joomla-cms
# [installation] Improve ajax error treatment

### Summary of Changes

There are some parts of the installation process that don't check for ajax request issues (timeout, syntax error, server error codes, etc) leaving the user with no clue on what is wrong.

This PR corrects that and also improves the error messages by replacing javascript alert messages with joomla message errors.

Didn't change the ftp part because don't have way to test it
### Testing Instructions
1. First apply this patch
2. Delete configuration.php and reload any site page you'll be redirect to installation
3. When in installation view: **Clear all browser cache**
4. Test no regresssions: normal installations and check everything is fine
5. Repeat step 2.
6. Test forcing errors, for checking you always get some error message:
   - Add an sql error in joomla.sql file. Try to install, you should get an error
   - Add some non php text (anything that generates a php syntax error) inside `/installation/model/configuration.php`. Try to install, you should get an error
   - Rename `/installation/controller/site.php` to anything. Try to install, you should get an error
   - Throw some exception (example: `throw new Exception('some error', 404);`) here https://github.com/joomla/joomla-cms/blob/staging/installation/controller/site.php#L34 . Try to install, you should get an error
   - Test forcing any other error you can remember in the install process
7. Code review. Use this to ignore whitespaces changes https://github.com/joomla/joomla-cms/pull/11930/files?w=1
### Documentation Changes Required

None.
